pkg/kubelet/cm/cpumanager/metrics_test.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cpumanager
+
+import (
+	"testing"
+
+	"github.com/blang/semver/v4"
+
+	"k8s.io/component-base/metrics/testutil"
+	"k8s.io/kubernetes/pkg/kubelet/cm/cpumanager/state"
+	"k8s.io/kubernetes/pkg/kubelet/cm/topologymanager"
+	"k8s.io/kubernetes/pkg/kubelet/metrics"
+	"k8s.io/utils/cpuset"
+)
+
+func TestMetricsAllocate(t *testing.T) {
+	testCases := []struct {
+		staticPolicyTest
+		expCSetAfterAlloc     cpuset.CPUSet
+		expCSetAfterRemove    cpuset.CPUSet
+		expCounterMetricValue float64
+	}{
+		{
+			staticPolicyTest: staticPolicyTest{
+				description: "SocketNoSMT",
+				topo:        topoUncoreDualSocketNoSMT,
+				pod: makeMultiContainerPod(
+					[]struct{ request, limit string }{
+						{"4000m", "4000m"}},
+					[]struct{ request, limit string }{
+						{"2000m", "2000m"}}),
+				containerName:   "initContainer-0",
+				stAssignments:   state.ContainerCPUAssignments{},
+				stDefaultCPUSet: cpuset.New(0, 1, 2, 3, 4, 5, 6, 7),
+			},
+			expCSetAfterAlloc:     cpuset.New(4, 5, 6, 7),
+			expCSetAfterRemove:    cpuset.New(2, 3, 4, 5, 6, 7),
+			expCounterMetricValue: 2,
+		},
+		{
+			staticPolicyTest: staticPolicyTest{
+				description: "SingleSocketHT",
+				topo:        topoSingleSocketHT,
+				pod: makeMultiContainerPod(
+					[]struct{ request, limit string }{
+						{"4000m", "4000m"}},
+					[]struct{ request, limit string }{
+						{"2000m", "2000m"}}),
+				containerName:   "initContainer-0",
+				stAssignments:   state.ContainerCPUAssignments{},
+				stDefaultCPUSet: cpuset.New(0, 1, 2, 3, 4, 5, 6, 7),
+			},
+			expCSetAfterAlloc:     cpuset.New(2, 3, 6, 7),
+			expCSetAfterRemove:    cpuset.New(1, 2, 3, 5, 6, 7),
+			expCounterMetricValue: 2,
+		},
+	}
+
+	metrics.ContainerAlignedComputeResources.Reset()
+	metrics.ContainerAlignedComputeResources.Create(&semver.SpecVersion)
```

Comment:
ok, we're inside a unit test and that's fine, but still I'd really like to make sure we can't possibly pollute any global state. Do we have any utility which ensure so (e.g. `testutil`?), or can't this ever be leaking into global state? If the latter please add a comment to explain

Reply:
Ah, I see. I'll either clear the metric if it's not created or restore it after the UT.